client/src/pages/MoodPage.jsx

```diff
@@ -21,22 +21,56 @@ ChartJS.register(
   Legend
 );
 
-export function calculateMoodStreak(moodLogs) {
-  if (!moodLogs || moodLogs.length === 0) return 0;
-
-  const datesSet = new Set(
-    moodLogs.map(entry => new Date(entry.createdAt).toDateString())
-  );
-
-  let streak = 0;
-  let currentDate = new Date();
+// Calculate streak correctly - must be consecutive days with no gaps
+export function calculateMoodStreak(logs) {
```

Comment:
Can you consolidate this function with the one in DashboardHome?
The logic is repeated in two files